src/levanter/models/backpack.py

```diff
@@ -161,13 +160,7 @@ def update_state_dict(self, state_dict: StateDict, prefix: Optional[str] = None)
         my_dict: StateDict = {}
         super().update_state_dict(my_dict, prefix)
 
-        es = cast(Axis, self.c_attn.In).size
-        num_heads = self.config.Senses.size
-        sense_head_size = self.config.SenseHeadDim.size
-
-        my_dict.update(
-            reshape_linear_layer(my_dict, apply_prefix(prefix, "c_attn"), (es,), (2 * num_heads * sense_head_size,))
-        )
+        my_dict.update(flatten_linear_layer(apply_prefix(prefix, "c_attn"), self.c_attn, out_dims_first_in_dict=True))
```

Comment:
Here's one error I got when running with this latest `flatten_linear_layer`:
```
Traceback (most recent call last):
  File "/home/ivanzhou/dev/levanter/examples/convert_backpack_to_hf.py", line 118, in <module>
    main()
  File "/home/ivanzhou/dev/levanter/src/levanter/config.py", line 585, in wrapper_inner
    response = fn(cfg, *args, **kwargs)
  File "/home/ivanzhou/dev/levanter/examples/convert_backpack_to_hf.py", line 94, in main
    _save_hf_checkpoint_local(
  File "/home/ivanzhou/dev/levanter/src/levanter/compat/hf_checkpoints.py", line 253, in _save_hf_checkpoint_local
    state_dict = model.to_state_dict()
  File "/home/ivanzhou/dev/levanter/src/levanter/compat/torch_serialization.py", line 50, in to_state_dict
    return jax_tree_to_state_dict(self, prefix)
  File "/home/ivanzhou/dev/levanter/src/levanter/compat/torch_serialization.py", line 118, in jax_tree_to_state_dict
    update_state_dict_with_jax_tree(tree, state_dict, prefix)
  File "/home/ivanzhou/dev/levanter/src/levanter/compat/torch_serialization.py", line 95, in update_state_dict_with_jax_tree
    tree.update_state_dict(state_dict, prefix)
  File "/home/ivanzhou/dev/levanter/src/levanter/models/backpack.py", line 370, in update_state_dict
    state_dict = super().update_state_dict(state_dict, prefix=prefix)
  File "/home/ivanzhou/dev/levanter/src/levanter/compat/torch_serialization.py", line 56, in update_state_dict
    return default_update_state_dict_with_eqx_module(state_dict, self, prefix)
  File "/home/ivanzhou/dev/levanter/src/levanter/compat/torch_serialization.py", line 160, in default_update_state_dict_with_eqx_module
    update_state_dict_with_jax_tree(v, state_dict, apply_prefix(prefix, k))
  File "/home/ivanzhou/dev/levanter/src/levanter/compat/torch_serialization.py", line 95, in update_state_dict_with_jax_tree
    tree.update_state_dict(state_dict, prefix)
  File "/home/ivanzhou/dev/levanter/src/levanter/models/backpack.py", line 164, in update_state_dict
    my_dict.update(flatten_linear_layer(apply_prefix(prefix, "c_attn"), self.c_attn, out_dims_first_in_dict=True))
  File "/home/ivanzhou/dev/levanter/src/levanter/compat/torch_serialization.py", line 173, in flatten_linear_layer
    weight = weight.flatten_axes(layer.Out, "__OUT__").flatten_axes(layer.In, "__IN__")
AttributeError: 'numpy.ndarray' object has no attribute 'flatten_axes'
```

When I inspect `self.c_attn` that is passed to `flatten_linear_layer` as linear, I found its weight and bias are numpy array instead of NamedArray:

```
Linear(
  weight=f32[1280,2,48,26](numpy),
  bias=f32[2,48,26](numpy),
  In=Axis(name='embed', size=1280),
  Out=(
    Axis(name='qk', size=2),
    Axis(name='senses', size=48),
    Axis(name='head_dim', size=26)
  )
)
```